### ISSUE TITLE
Tested masked are all zeros.

### ISSUE THREAD

**Issue by August-us** [milesial/Pytorch-UNet#101]
Thanks for share your code. However, when I used it, I got a error make me very confused. That is I use the transforms.Resize(full_img.size[1]), and the (probs==0).all() is True. So far, it is obviously impossible to get the correct calculation results. But before I transofrm, If I print the probs.sum(axis=0), we get the result is all ones. Does it indicate background in the classfication results? Should I set the n_calsses to category +background. The first question is particaluarly important. I hope somone who knows helps me to slove it soon, appreciate it.

**Comment by milesial** [comment]
If `probs.sum(axis=0)` is all 1s then it is correct if you are using 2 classes or more.
If you have one category + background, you could set `n_classes` to 1 or 2, it is equivalent. (with softmax, the second category is determined by the first).

With more classes, it depends on your interpretation of your problem. If you wish to predict "background" for some pixels, then obviously you should treat it as a class.

**Comment by August-us** [comment]
> `probs.sum(axis=0)` is all 1s then it is correct if you are using 2 classes or more.
> If you have one category + background, you could set `n_classes` to 1 or 2, it is equivalent. (with softmax, the second category is determined by the first).
> 
> With more classes, it depends on your interpretation of your problem. If you wish to predict "background" for some pixels, then obviously you should treat it as a class.

Thanks for your replay, You solved my second question. And I want to know that I use the Resize and defult args interpolation=Image.BILINEAR. Although all categories are predicted as background, if I resize the image to full_size. I would get a zeros imags.

**Comment by milesial** [comment]
If all pixels are background, after the thresholding then it is normal that you get all zeros (for a binary problem). If you remove the thresholding, then you should have a grey-level output mask. I don't think it depends on the resizing but on the thresholding.

**Comment by August-us** [comment]
Soory, I made a mistake, I got the zeros when I run the full_mask = dense_crf(np.array(full_img).astype(np.uint8), full_mask). Appreciate your help.

**Comment by milesial** [comment]
oh, just can just keep the dense CRF disabled, as it is by default, it is just an experimental feature that does not add much to the accuracy.
